src/Framework/Gacela.php

```diff
@@ -155,4 +162,53 @@ private static function runPlugins(Config $config): void
             self::$mainContainer->resolve($current);
         }
     }
+
+    private static function addModuleBindingsToSetup(SetupGacelaInterface $setup): void
+    {
+        $setup->combine(SetupGacela::fromCallable(static function (GacelaConfig $config): void {
+            foreach (self::collectBindingsFromProviders() as $k => $v) {
+                $config->addBinding($k, $v);
+            }
+        }));
+    }
+
+    /**
+     * @return array<class-string, class-string|callable|object>
+     */
+    private static function collectBindingsFromProviders(): array
+    {
+        if (self::$appRootDir === null || !is_dir(self::$appRootDir)) {
+            return [];
+        }
+
+        $iterator = new RecursiveIteratorIterator(
+            new RecursiveDirectoryIterator(self::$appRootDir),
+        );
+
+        $result = [];
+        /** @var SplFileInfo $file */
+        foreach ($iterator as $file) {
+            if ($file->getExtension() === 'php') {
+                $fileContents = (string)file_get_contents($file->getPathname());
+                if (preg_match('/namespace\s+([a-zA-Z0-9_\\\\]+)\s*;/', $fileContents, $matches) !== false) {
```

Comment:
What if the file has no namespace on top BUT has some var (varName or varValue) that contains the keyword "namespace"?
Have you checked this case?